json-defined-input-formats.md

```diff
@@ -0,0 +1,75 @@
+# JSON-Defined Input Formats
+Shesmu olives can access user-defined input formats. Normally, data is provided from Java plugins,
+so the input format itself is also defined in Java. If that is the case,
+consult [Plugin Implementation Guide](implementation.md) to see how to do that. If a plugin is not
+required, it is possible to define an input format without Java. This format can consume input from
+JSON, either using a local file or from a remote endpoint.
+
+Unlike other Shesmu configuration files, JSON-defined input formats are read _only on startup_.
```

Comment:
Exactly one shot at reading the file during start up and then it never looks again at the file or for additional files.

Basically, the compiler bakes a bunch of input format data into global state, so if it were to change, everything would be come a nightmare race condition. This is fine with the Java input formats, because there isn't a way to add a new class after start up. For these, they have to be treated the same way.

Should I just paste in this explanation?

Reply:
Sounds like a plan, thanks

Reply:
Done.